manifests/extension/config.pp

```diff
@@ -102,19 +107,48 @@
   $ext_tool_enabled  = pick_default($::php::ext_tool_enabled, $::php::params::ext_tool_enabled)
 
   if $facts['os']['family'] == 'Debian' and $ext_tool_enabled {
-    $cmd = "${ext_tool_enable} -s ${sapi} ${so_name}"
+    case $so_name {
+      'mysql': {
+        case $php::globals::php_version {
+          /^7\.[0-9]$/: {
+            $cmds = {
+              'mysqlnd' => "${ext_tool_enable} -s ${sapi} mysqlnd",
+              'mysqli' => "${ext_tool_enable} -s ${sapi} mysqli",
+              'pdo_mysql' => "${ext_tool_enable} -s ${sapi} pdo_mysql",
+            }
+          }
+          default: {
+            $cmds = {
+              'mysql' => "${ext_tool_enable} -s ${sapi} mysql",
+              'mysqli' => "${ext_tool_enable} -s ${sapi} mysqli",
+              'pdo_mysql' => "${ext_tool_enable} -s ${sapi} pdo_mysql",
+            }
+          }
+        }
+      }
+      default: {
+        $cmds = {
+          $so_name => "${ext_tool_enable} -s ${sapi} ${so_name}",
+        }
+      }
+    }
 
     $_sapi = $sapi? {
       'ALL' => 'cli',
       default => $sapi,
     }
-    exec { $cmd:
-      onlyif  => "${ext_tool_query} -s ${_sapi} -m ${so_name} | /bin/grep 'No module matches ${so_name}'",
-      require => ::Php::Config[$title],
-    }
 
-    if $::php::fpm {
-      Package[$::php::fpm::package] ~> Exec[$cmd]
+    # run cmds. normally only one command is used for one extension, but debian split up the mysql extension, which now
+    # needs more commands to run.
+    $cmds.each |String $my_so_name, String $my_cmd| {
+      exec { $my_cmd:
+        onlyif  => "${ext_tool_query} -s ${_sapi} -m ${my_so_name} | /bin/grep 'No module matches ${my_so_name}'",
+        require => ::Php::Config[$title],
+      }
+
+      if $::php::fpm {
```

Comment:
can you remove the trailing ::, while you are already refactoring? They are totally unneeded.

Reply:
i would suggest to do this (i can do it) in a seperate PR, i personally dislike having noisy diffs from style fixes together with logical changes.